pkg/cloud/services/network/subnets.go

```diff
@@ -398,6 +402,10 @@ func (s *Service) describeSubnets() (*ec2.DescribeSubnetsOutput, error) {
 }
 
 func (s *Service) createSubnet(sn *infrav1.SubnetSpec) (*infrav1.SubnetSpec, error) {
+	// When managing subnets, the ID specified in the spec is the name of the subnet.
+	sn.Tags["Name"] = sn.ID
```

Comment:
Need to cater for an empty map here, panics otherwise
```suggestion
	if sn.Tags == nil {
		sn.Tags = make(infrav1.Tags)
	}
	sn.Tags["Name"] = sn.ID
```